webservices/phoenix_deployment.md

```diff
@@ -0,0 +1,192 @@
+
+There are two ways of handling this.
+
+### Compile time
+
+For compile time env vars, you need to ensure the values you need are loaded into the build server's environment.
+
+```bash
+# ~/.profile
+
+..
+export CONFIG_VALUE=foo
+..
+```
+note: `~/.profile` is automatically loaded into the environment during edeliver operations, so it's a good place to store environment variables on the build machine.
+
+Now if you have a config definition like this:
+
+```elixir
+# prod.exs
+
+..
+config :deployment_demo, config_val: System.get_env("CONFIG_VALUE")
+..
+```
+
+This will fetch the value from the environment at compile time and have the value present when attempting to run
+```elixir
+Application.get_env(:deployment_demo, :config_val)
+# "foo"
+```
+
+If the variable is not present at compile time, during runtime when `get_env` is called, it will return `nil`.
+```elixir
+Application.get_env(:deployment_demo, :config_val)
+# nil
+```
+
+### Runtime
+
+If you wish to only store environment variables on the production server, you need to ensure that you're actually fetching from the environment at runtime.
+
+The easiest way of accomplishing this, is to reference environment variables via functions. For example:
+
+Let's take the example from before, but instead of calling `System.get_env()` in the config file, let's define the environment key we want to use in the config, but access the environment at runtime.
+
+```elixir
+# prod.exs
+
+..
+config :deployment_demo, config_val: "CONFIG_VALUE"
+..
+```
+
+Now in our module we can access the value via a function.
+
+```elixir
+# my_module.ex
+
+defmodule DeploymentDemo.MyModule do
+..
+  defp config_value do
+    :deployment_demo
+    |> Application.get_env(:config_val)
+    |> System.get_env()
+  end
+end
+```
+
+This can easily be cleaned up by extracting it to a config helper module.
+
+```elixir
+# config_helper.ex
+
+defmodule DeploymentDemo.ConfigHelper do
+  def get_env(app, value) do
+    app
+    |> Application.get_env(value)
+    |> System.get_env()
+  end
+end
+```
+
+then you can simply alias the `ConfigHelper` module and fetch values from the environment via
+```elixir
+ConfigHelper.get_env(:deployment_demo, :config_val)
+# "foo"
+```
+
+## Our first release
+
+Now that we have distillery setup and building releases and edelivery hooked up to our build and production servers, we're ready to run our first release.
+
+From your local machine, from the app's root directory. Run the commands:
+```
+$ mix edeliver build release
+$ mix edeliver deploy release to production
+```
```

Comment:
I think adding the start/stop/ping commands would be useful here

Reply:
added start/ping, stop doesn't make sense in this context at least, but added a link to some of the other commands